docs/src/tutorials/getting_started/transitioning_from_matlab.jl

```diff
@@ -0,0 +1,280 @@
+
+# Like YALMIP and CVX, JuMP allows you to recover the dual variables. In order
+# to do that, the simplest method is to name the constraint you're interested in,
+# for example, `bob = @constraint(model, sum(v) == 1)` and then, after the
+# optimzation is done, call `dual(bob)`.
+
+# See [Duals of variable bounds](@ref) for more.
+
+# ## Reformulating problems
+
+# Perhaps the biggest difference between JuMP and YALMIP and CVX is how far the
+# modeller is willing to go in reformulating the problems you give to it. CVX is
+# happy to reformulate anything it can, even using approximations if your solver
+# cannot handle the problem. YALMIP will only do exact reformulations, but is
+# still fairly adventurous, being willing to reformulate a nonlinear objective
+# in terms of conic constraints, for example. JuMP does no such thing: it only
+# reformulates objectives into objectives, and constraints into constraints, and
+# is fairly conservative at that. As a result, you might need to do some
+# reformulations manually, for which a good guide is available [here](@ref conic_tips_and_tricks).
+
+# An interesting possibility that JuMP offers is turning off reformulations
+# altogether, which will reduce latency if they're in fact not needed, or
+# produce an error if they are. To do that, set your optimizer with the option
+# `add_bridges = false`, as for example, in `set_optimizer(model, Hypatia.Optimizer, add_bridges=false)`
+
+# ## Vectorization
+
+# In MATLAB it is absolutely essential to "vectorize" your code to obtain
+# acceptable performance. This is because MATLAB is a very slow interpreted
+# language, which sends your commands to extremely fast libraries. When you
+# "vectorize" your code you're minimizing the MATLAB part of the work and sending
+# it to the libraries instead. There's no such duality with Julia. Everything you
+# write and most libraries you use will compile down to LLVM, so "vectorization"
+# has no effect.
+
+# For example, if you are writing a linear program in MATLAB and instead of the
+# usual `constraints = [v >= 0]` you write:
+# ```matlab
+# for i = 1:n
+#    constraints = [constraints, v(i) >= 0];
+# end
+# ```
+# performance will be horrible. With Julia, on the other hand, there's hardly
+# any difference between `@constraint(model, v >= 0)` and
+
+for i in 1:n
+    @constraint(model, v[i] >= 0)
+end
+
+# ## Rosetta stone
+
+# To finish this tutorial, we show a complete example of the same optimization
+# problem being solved with JuMP, YALMIP, and CVX. It is an SDP computing a
+# simple lower bound on the random robustness of entanglement using the partial
+# transposition criterion. The code is complete apart from the function that does
+# partial transposition. With JuMP we use the function `partialtranspose` from
+# [Convex.jl](https://jump.dev/Convex.jl/stable/). With both YALMIP and CVX we
+# use the function `PartialTranspose` from [QETLAB](https://github.com/nathanieljohnston/QETLAB).
```

Comment:
Using `PartialTranspose` is too complicated. We want something that works out-of-the-box for JuMP and MATLAB.

Reply:
I really want to use this example for three reasons:

1. It illustrates a major advantage of JuMP over Convex, in that you can use whatever functions you want to build your problems, you don't need them to be specifically coded for JuMP. This is why I gave up using CVXPY, fwiw.
2. It tells people where to find a good implementation of partial transposition. It is the thing everyone asks me when I tell them to switch to Julia.
3. It is the canonical example I use to teach duality in my classes and in my (review)[https://arxiv.org/abs/2307.02551]